runbotics-orchestrator/src/main/java/com/runbotics/web/rest/UserBotResource.java

```diff
@@ -0,0 +1,109 @@
+package com.runbotics.web.rest;
+
+import com.runbotics.domain.UserBot;
+import com.runbotics.repository.UserBotRepository;
+import com.runbotics.service.UserBotService;
+import com.runbotics.service.dto.UserBotDTO;
+import com.runbotics.web.rest.errors.BadRequestAlertException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import tech.jhipster.web.util.HeaderUtil;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class UserBotResource {
+
+    private static final String ENTITY_NAME = "jhi_user_bot";
+    private final Logger log = LoggerFactory.getLogger(UserBotResource.class);
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+    private final UserBotService userBotService;
+    private final UserBotRepository userBotRepository;
+
+    public UserBotResource(UserBotService userBotService, UserBotRepository userBotRepository) {
+        this.userBotService = userBotService;
+        this.userBotRepository = userBotRepository;
+    }
+
+    /**
+     * {@code GET /bot-notifications/{botId}} : gets all subscriptions for botId.
+     *
+     * @param botId bot id to get all filtered subscriptions.
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the subscriptions.
+     * @throws BadRequestAlertException {@code 400 (Bad Request)} if the botId was not provided.
+     */
+    @GetMapping("/bot-notifications/{botId}")
+    public ResponseEntity<List<UserBotDTO>> getAllBotSubscriptions(@Valid @PathVariable Long botId) {
+        log.debug("REST request to get all subscriptions for bot notification with id: {}", botId);
+
+        if (botId == null) {
+            throw new BadRequestAlertException("To get all bot subscriptions botId must be provided", ENTITY_NAME, "idNotExists");
+        }
+
+        List<UserBotDTO> result = userBotService.getAllSubscriptionsByBotId(botId);
+
+        return ResponseEntity.ok().body(result);
+    }
+
+    /**
+     * {@code POST /bot-notifications} : creates a new subscription for bot notification.
+     *
+     * @param userBotDTO the subscription to create.
+     * @return the {@link ResponseEntity} with status {@code 201 (Created)} and with body the new subscription.
+     * @throws BadRequestAlertException {@code 400 (Bad Request)} if the userId or botId was not provided.
+     */
+    @PostMapping("/bot-notifications")
+    public ResponseEntity<UserBotDTO> createBotSubscription(@Valid @RequestBody UserBotDTO userBotDTO) {
```

Comment:
@Valid is not neccessarry, because you haven't used any 'valid' annotations in Entity.
You can remove it OR remove both belowed if conditions, and use some annotations above fields of entity, f.e.:
    @NotBlank(message = "UserId is mandatory")
    userId;
    
    or in this case:
    @NotNull(message = "To create new bot subscription botId must be provided")
    private double botId;
    
it's from: import jakarta.validation.constraints.NotNull;

but imo you can just remove this annotation, I wrote this comment for your knowledge  :)